lib/libbluetooth/bluetooth.c

```diff
@@ -101,29 +101,42 @@ struct hostent *
 bt_gethostent(void)
 {
 	char	*p, *cp, **q;
+	int retry_count = 0; // Retry counter to prevent infinite loops
 
 	if (hostf == NULL)
 		hostf = fopen(_PATH_BT_HOSTS, "r");
 
 	if (hostf == NULL) {
 		h_errno = NETDB_INTERNAL;
+		perror("fopen"); // Log the error
 		return (NULL);
 	}
-again:
-	if ((p = fgets(buf, sizeof(buf), hostf)) == NULL) {
-		h_errno = HOST_NOT_FOUND;
+
+	while (retry_count < 100) { // Limit retries to 100
+		if ((p = fgets(buf, sizeof(buf), hostf)) == NULL) {
+			h_errno = HOST_NOT_FOUND;
+			return (NULL);
+		}
+		if (*p == '#')
```

Comment:
Looks like you limited number of comment lines to 100. That is wrong although having a such amount of comments in hosts file looks strange